types/module/module.go

```diff
@@ -143,21 +156,39 @@ func (bm BasicManager) AddQueryCommands(rootQueryCmd *cobra.Command) {
 // AppModuleGenesis is the standard form for an application module genesis functions
 type AppModuleGenesis interface {
 	AppModuleBasic
+	LegacyGenesis
+}
 
+// LegacyGenesis is the legacy interface for stateful genesis methods.
+type LegacyGenesis interface {
+	LegacyGenesisBasics
 	InitGenesis(sdk.Context, codec.JSONCodec, json.RawMessage) []abci.ValidatorUpdate
 	ExportGenesis(sdk.Context, codec.JSONCodec) json.RawMessage
 }
 
-// AppModule is the standard form for an application module
+// Deprecated: AppModule is the legacy form for an application module. Most of
+// its functionality has been moved to extension interfaces based off of
+// cosmossdk.io/core/appmodule.AppModule.
 type AppModule interface {
-	AppModuleGenesis
+	appmodule.AppModule
+
+	AppModuleBasic
+}
 
-	// registers
+// HasRegisterInvariants is the interface for registering invariants.
+type HasRegisterInvariants interface {
+	// RegisterInvariants registers module invariants.
 	RegisterInvariants(sdk.InvariantRegistry)
+}
 
-	// RegisterServices allows a module to register services
+// LegacyRegisterServices is the legacy method for modules to register services.
+type LegacyRegisterServices interface {
```

Comment:
This is legacy because of the new core api?

Reply:
Yes. I have the idea to push the new core API version relatively quickly after this. Or we could do the rename to legacy once the core API version is ready. Don't have a strong preference just thought this was clearer.

Reply:
the import will change to use the new api, correct?